Work around hot-reloading issue

sharils opened the pull request:
See https://github.com/phoenixframework/phoenix/issues/1165 for more detail

<!--
READ THIS: Before creating a pull request, please consider the following first.

* The most important thing you can do is write tests. Code without tests
  probably doesn't work, and will almost certainly stop working later on. Pull
  requests without tests probably won't be accepted, although there are some
  exceptions.
* Read the Contributing guide linked above first.
* If you are adding a new linter, remember to update the README.md file and
  doc/ale.txt first.
* If you add or modify a function for converting error lines into loclist items
  that ALE can work with, please add Vader tests for them. Look at existing
  tests in the test/handler directory, etc.
* If you add or modify a function for computing a command line string for
  running a command, please add Vader tests for that.
* Generally try and cover anything with Vader tests, although some things just
  can't be tested with Vader, or at least they can be hard to test. Consider
  breaking up your code so that some parts can be tested, and generally open up
  a discussion about it.
* Have fun!
-->

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/w0rp/ale/1266)
<!-- Reviewable:end -->

w0rp commented on ale_linters/elixir/credo.vim:
```diff
@@ -32,6 +32,6 @@ endfunction
 \   'name': 'credo',
 \   'executable': 'mix',
-\   'command': 'mix credo suggest --format=flycheck --read-from-stdin %s',
+\   'command': 'MIX_ENV=lint mix credo suggest --format=flycheck --read-from-stdin %s',
```
For windows, this will need to be `set MIX_ENV=lint && ...` Look at say `ale_linters/slim/slimlint.vim` for an example of setting environment variables for all supported platforms.

sharils replied:
Thanks @w0rp . But I'll need more time to incorporate that and test it on windows. Feel free to close this PR if you'd like to. I can open another one when it's ready.

sharils commented:
Hi @w0rp - Would you mind reviewing again? I've just rebased and merged my latest commit because the first commit caused other issue, and I have to take another approach and it turns out that approach doesn't have to rely on setting environment variables. Since the intermediate approach doesn't work, I just removed that commit. Hope you don't mind. Thanks a million.

w0rp commented:
Cheers! :beers:
